mergin/client_pull.py

```diff
@@ -551,3 +572,176 @@ def pull_project_finalize(job):
 
     shutil.rmtree(job.temp_dir)
     return conflicts
+
+
+def download_file_async(mc, project_path, file_path, output_file, version):
+    """
+    Starts background download project file at specified versions.
+    Returns handle to the pending download.
+    """
+    mc.log.info(f"--- start download {file_path} for {project_path}")
+    project_info = mc.project_info(project_path, version=version)
+
+    mc.log.info(f"Got project info. version {project_info['version']}")
+
+    # set temporary directory and make sure the destination directory exists
+    temp_dir = os.path.join(tempfile.gettempdir(), "mergin_temp")
+    os.makedirs(os.path.dirname(output_file), exist_ok=True)
+
+    download_list = []
+    update_tasks = []
+    total_size = 0
+    for file in project_info['files']:
+        if file["path"] == file_path:
+            file['version'] = version
+            items = _download_items(file, temp_dir)
+            task = UpdateTask(file['path'], items, output_file)
+            download_list.extend(task.download_queue_items)
+            for item in task.download_queue_items:
+                total_size += item.size
+            update_tasks.append(task)
+            break
+    if not download_list:
+        warn = f"No {file_path} exists at version {version}"
+        mc.log.warning(warn)
+        raise ClientError(warn)
+
+    mc.log.info(f"will download file {file_path} in {len(download_list)} chunks, total size {total_size}")
+    job = DownloadJob(
+        project_path, total_size, version, update_tasks, download_list, temp_dir, None, mc, project_info
```

Comment:
mp is None? if we have access to MerginProject in the end, let's pass it - lower complexity